File: Python/m00_foundations/ch02_data_input_output/output_formatting.py
# Inputs: name, age, weight
name = input('Enter your name: ')
age = int(input('Enter your age: '))
weight = float(input('Enter your weight: '))
print()

# Example 1 = Basic Concatenation:
print('1. My name is', name, 'I am', age, 'years old and I weigh', weight, 'Kg')

# Example 2 = Using str.format() method:
# Basic formatting
print('2. My name is {} I am {} years old and I weigh {} Kg'.format(name, age, weight))
# Formatting by specifying the position of variables
print('2. My name is {0} I am {1} years old and I weigh {2:.1f} Kg'.format(name, age, weight))
# Formatting explicit naming of arguments
print('2. My name is {n} I am {i} years old and I weigh {p:.2f} Kg'.format(n=name, i=age, p=weight))

# Example 3 = Using f-strings (Formatted String Literals):
# Including expressions inside f-strings
print(f'3. My name is {name} I am {age} years old and I weigh {weight:.2f} Kg')

File: JavaScript/m00-foundations/ch02-data-input-output/outputFormatting.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

/**
 * Different ways to format output in JavaScript.
 *
 * Prompts the user to enter their name, age and weight and then
 * displays the entered values in three different ways:
 *
 * 1. Using basic concatenation.
 * 2. Using template literals and formatting numbers.
 * 3. Using template literals with expressions.
 *
 * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals
 *
 * @function main
 */
function main() {
  console.log('---------------------');
  console.log('Output Formatting Program');
  console.log('---------------------');
  console.log();

  rl.question('Enter your name: ', (name) => {
    rl.question('Enter your age: ', (age) => {
      rl.question('Enter your weight: ', (weight) => {
        console.log();

        // Example 1 = Basic Concatenation:
        console.log('1. My name is ' + name + ', I am ' + age + ' years old and I weigh ' + weight + 'Kg');
        // Example 2 = Using Template Literals and formatting numbers:
        console.log(`2. My name is ${name}, I am ${age} years old and I weigh ${parseFloat(weight).toFixed(2)}Kg`);
        // Example 3 = Using Template Literals with expressions:
        console.log(`3. My name is ${name}, I am ${age} years old and I weigh ${weight}Kg`);

        rl.close();
      });
    });
  });
}

main();
